SAKPaaS/src/app/components/global-dialog/global-dialog.component.ts

```diff
@@ -21,14 +21,18 @@ export class GlobalDialogComponent {
   ) { }
 
   onOkay(): void {
-    if (this.cookieCheckbox && this.gpsCheckbox) {
-      this.dialogRef.close(DialogMessageReturnTypes.OKAY);
-    } else if (this.cookieCheckbox && !this.gpsCheckbox) {
-      this.dialogRef.close(DialogMessageReturnTypes.ONLYCOOKIES);
-    } else if (!this.cookieCheckbox && this.gpsCheckbox) {
-      this.dialogRef.close(DialogMessageReturnTypes.ONLYGPS);
+    if (this.data.askForPermission) {
+      if (this.cookieCheckbox.value && this.gpsCheckbox.value) {
```

Comment:
Ich hätte hier das Interface `CheckboxesDialog`, was du ja schon definiert hast, gleich verwendet. Dann kannst du in dem if-Fall einfach
```ts
this.dialogRef.close(
    {
        cookiesAllowed: this.cookieCheckbox.value,
        gpsAllowed: this.gpsCheckbox.value
    }
);
```
machen und müsstest das nicht zweimal hin und zurück bobbeln. Das müsstest du dann auch an der Stelle anpassen, wo du den Rückgabewert von dem Dialog empfängst.

Vielleicht wäre es tatsächlich besser, das in eine eigene Komponente zu machen (also alles, was du jetzt in dem GlobalDialogComponent gemacht hast rauskopieren und einen PermissionDialog oder sowas draus machen) @nclskfm ?

Reply:
> Vielleicht wäre es tatsächlich besser, das in eine eigene Komponente zu machen (also alles, was du jetzt in dem GlobalDialogComponent gemacht hast rauskopieren und einen PermissionDialog oder sowas draus machen) @nclskfm ?
@timgrohmann ja hab ich mir auch schon überlegt. Aber dann kann man `GlobalDialogComponent` auch entfernen, weil das ist aktuell der einzige Anwendungsfall.^^

Reply:
Dann halt umbenennen und den ursprünglichen Code entfernen :)

Reply:
> Ich hätte hier das Interface `CheckboxesDialog`, was du ja schon definiert hast, gleich verwendet. Dann kannst du in dem if-Fall einfach
> 
> ```ts
> this.dialogRef.close(
>     {
>         cookiesAllowed: this.cookieCheckbox.value,
>         gpsAllowed: this.gpsCheckbox.value
>     }
> );
> ```
> 
> machen und müsstest das nicht zweimal hin und zurück bobbeln. Das müsstest du dann auch an der Stelle anpassen, wo du den Rückgabewert von dem Dialog empfängst.
> 
> Vielleicht wäre es tatsächlich besser, das in eine eigene Komponente zu machen (also alles, was du jetzt in dem GlobalDialogComponent gemacht hast rauskopieren und einen PermissionDialog oder sowas draus machen) @nclskfm ?

@timgrohmann Dann müsste aber glaube ich das Observable von showDialog `CheckboxesDialog` sein, was (falls GlobalDialog nicht zu PermissionDialog wird) glaube ich nicht so sinnvoll ist.

Reply:
Ja, das stimmt. Also ich wäre dafür, das für deinen einen Usecase zu spezifizieren, weil da ist ja wirklich mehr Funktionalität und State drin, als ein allgemeiner Dialog braucht.